src/components/Dropdown.tsx

```diff
@@ -0,0 +1,97 @@
+import classNames from 'classnames';
+import React, { useState, useMemo, useCallback } from 'react';
+
+import { Person } from '../types/Person';
+import { peopleFromServer } from '../data/people';
+
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timerId = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+type Props = {
+  onSelected: (person: Person | null) => void;
+  delay: number;
+};
+
+export const Dropdown: React.FC<Props> = ({ onSelected, delay }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [delay]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleButtonClick = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+    onSelected(person);
+    setQuery(person.name);
+  };
+
+  const hasDropdown = React.useMemo(() => {
```

Comment:
No need to use `React.useMemo`. You can use directly `useMemo` as you did for other places.